packages/react/src/primitives/Message/Message.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { MessageHeading } from './MessageHeading';
+import { MessageIcon } from './MessageIcon';
+import { MessageDismiss } from './MessageDismiss';
+import { MessageContent } from './MessageContent';
+import { MessageContainer } from './MessageContainer';
+
+import {
+  MessageProps,
+  BaseMessageProps,
+  ForwardRefPrimitive,
+  Primitive,
+} from '../types';
+
+const MessagePrimitive: Primitive<MessageProps, 'div'> = (
+  {
+    children,
+    heading,
+    dismissButtonLabel,
+    isDismissible,
+    onDismiss,
+    hasIcon = true,
+    colorTheme = 'neutral',
+    variation = 'filled',
+    ...rest
+  },
+  ref
+) => {
+  return (
+    <MessageContainer
+      colorTheme={colorTheme}
+      variation={variation}
+      ref={ref}
+      {...rest}
+    >
+      {hasIcon ? <MessageIcon /> : null}
+      <MessageContent>
+        {heading ? <MessageHeading>{heading}</MessageHeading> : null}
+        {children}
+      </MessageContent>
+
+      {isDismissible ? (
+        <MessageDismiss
+          onDismiss={onDismiss}
+          dismissButtonLabel={dismissButtonLabel}
+        />
+      ) : null}
+    </MessageContainer>
+  );
+};
+
+/**
+ * [📖 Docs](https://ui.docs.amplify.aws/react/components/message)
+ */
+
+type MessageType = ForwardRefPrimitive<BaseMessageProps, 'div'> & {
+  Container: typeof MessageContainer;
+  Content: typeof MessageContent;
+  Dismiss: typeof MessageDismiss;
+  Icon: typeof MessageIcon;
+  Heading: typeof MessageHeading;
```

Comment:
Let's use concrete types here

Reply:
Sorry what do you mean by concrete types?

Reply:
I did this for Breadcrumbs. We could probably do something like `Container: ForwardRefPrimitive<BaseMessageContainerProps, 'div'>` but that would just be copying the type from the MessageContainer. Doing the typeof thing seems to work for intellisense and props table generation and whatnot